### PR TITLE
refactor: read installationID from new location

### PR DESCRIPTION
**Pull request by frrist**
- must follow: https://github.com/bacalhau-project/getbacalhauorg/pull/37

### CODE REVIEW

File: pkg/repo/fs.go [hunk 1]
```diff
@@ -98,15 +98,20 @@ func (fsr *FsRepo) Init() error {
...
-	}
-
 	if err := fsr.WriteVersion(Version4); err != nil {
 		return fmt.Errorf("failed to persist repo version: %w", err)
 	}
 
+	// these calls must happen after version write for the file to exist
+	if err := fsr.WriteInstallationID(ReadInstallationID()); err != nil {
+		log.Trace().Err(err).Msg("failed to write installationID")
+	}
```
Comment (wdbaruni) [inline pkg/repo/fs.go:108]: why are we storing the installationID in the repo?

File: pkg/repo/fs.go [hunk 2]
```diff
@@ -135,6 +140,16 @@ func (fsr *FsRepo) Open() error {
...
+	// check if an installationID exists attempting to persist one if not found.
+	// never fail here as this isn't critical to node start up.
+	if installationID, err := fsr.ReadInstallationID(); err != nil {
+		log.Trace().Err(err).Msgf("failed to read installationID")
+	} else if installationID == "" {
+		if err := fsr.WriteInstallationID(ReadInstallationID()); err != nil {
+			log.Trace().Err(err).Msg("failed to write installationID")
+		}
+	}
+
```
Comment (wdbaruni) [inline pkg/repo/fs.go:152]: why are we persisting one if not found? As we discussed before, installationID should only be provided by the installation script and we won't generate one on our own when running bacalhau. That will be handled by the instanceID

File: pkg/repo/installationid.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		}
+		idFile = filepath.Join(configDir, "bacalhau", "installation_id")
+	case "darwin":
+		idFile = filepath.Join(os.Getenv("HOME"), "Library", "Preferences", "com.bacalhau.installation_id")
+	case "windows":
+		appData := os.Getenv("APPDATA")
+		idFile = filepath.Join(appData, "bacalhau", "installation_id")
+	default:
+		configDir := filepath.Join(os.Getenv("HOME"), ".config")
+		idFile = filepath.Join(configDir, "bacalhau", "installation_id")
```
Comment (wdbaruni) [inline pkg/repo/installationid.go:27]: lets use this for mac as well

File: pkg/repo/installationid.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+func ReadInstallationID() string {
+	var idFile string
+	os.UserCacheDir()
```
Comment (wdbaruni) [inline pkg/repo/installationid.go:12]: not used

**Review by wdbaruni — CHANGES_REQUESTED** [review]

File: pkg/repo/installationid.go (hunk 3 above)
Comment (frrist) [reply pkg/repo/installationid.go:24]: we do on line 13

**Comment by frrist** [comment]
closing in favor of https://github.com/bacalhau-project/bacalhau/pull/4450

**Pull request closed without merging**